packages/registrar_client/src/utils.ts

```diff
@@ -2,23 +2,40 @@ import * as base64 from "base64-js";
 import * as tweetnacl from "tweetnacl";
 import { AxiosRequestConfig } from "axios";
 import { Buffer } from "buffer";
+import { mnemonicToSeed, validateMnemonic} from "bip39";
 
 function createSignatureForChallenge(challenge: string, privateKey: string): string {
   const signature = tweetnacl.sign.detached(Buffer.from(challenge, "utf-8"), base64.toByteArray(privateKey));
   return base64.fromByteArray(signature);
 }
 
-export function createSignatureWithPublicKey(publicKey: string, privateKey: string): { signature: string; timestamp: number } {
-  if (publicKey === "") {
-    throw new Error("Public key is required");
+async function deriveKeyPair(mnemonicOrSeed: string): Promise<{ publicKey: string; privateKey: string; }> {
+  let seed : Buffer;
+  if (validateMnemonic(mnemonicOrSeed)) {
+    seed = (await mnemonicToSeed(mnemonicOrSeed)).subarray(0, 32);
+
+  } else if(mnemonicOrSeed.startsWith("0x")) {
+    seed = Buffer.from(mnemonicOrSeed.slice(2), "hex"); 
+  } else {
+    throw new Error("Invalid seed or mnemonic");
   }
+  const keyPair = tweetnacl.sign.keyPair.fromSeed(seed);
+  return {
+    publicKey: base64.fromByteArray(keyPair.publicKey),
+    privateKey: base64.fromByteArray(keyPair.secretKey),
+  };
+}
```

Comment:
The way the keypair is derived won't be compatible with the other clients because they use Polkadot or Subkey tools.

- So let's use [polkadot/keyring](https://www.npmjs.com/package/@polkadot/keyring)
- [Here](https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/tfchain_client/src/client.ts#L321) is an usage example
- `bip39` can still be used for mnemonic generation
- Support keypair types `sr25519` and `ed25519`. (default: `sr25519`)